File: src/javabot/operations/IgnoreOperation.java
package javabot.operations;

import com.rickyclarkson.java.lang.Debug;

import java.util.ArrayList;
import java.util.List;
import com.rickyclarkson.java.util.TypeSafeList;
import javabot.BotEvent;
import javabot.Message;

/**
	@deprecated weird
*/
public class IgnoreOperation implements BotOperation {
    public List handleMessage(BotEvent event) {
	    Debug.printDebug("About to exit");
	    System.exit(0);
        List messages = new TypeSafeList(new ArrayList(), Message.class);
        String message = event.getMessage();
        //Javabot bot = event.getBot();
        String[] parts = message.split(" ");
/*        if((parts.length == 3)
            && parts[0].equals("quit ")
            && parts[2].equals(bot.getNickPassword())) {

            //bot.addIgnore(parts[1]);
        }*/
        return messages;
    }

    public List handleChannelMessage(BotEvent event)
    {
	    	return new TypeSafeList(new ArrayList(),Message.class);
    }
}

File: src/main/kotlin/javabot/operations/IgnoreOperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import com.google.inject.Inject
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao

class IgnoreOperation @Inject constructor(bot: Javabot, adminDao: AdminDao) :
    BotOperation(bot, adminDao) {
    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value
        if (message.startsWith("ignore ")) {
            val parts = message.split(" ")
            bot.addIgnore(parts[1])
            responses.add(Message(event, Sofia.botIgnoring(parts[1])))
        }
        return responses
    }
}
